File: src/main/java/lec12/Lec12Main2.java
package lec12;

public class Lec12Main2 {
    public static void main(String[] args) {
        moveSomething(new Movable() {

            @Override
            public void move() {
                System.out.println("움직인다~~");
            }

            @Override
            public void fly() {
                System.out.println("난다~~");
            }
        });
    }

    private static void moveSomething(Movable movable) {
        movable.move();
        movable.fly();
    }
}

File: src/main/kotlin/lec12/Lec12Main2.kt
package lec12

fun main() {
    moveSomething(object : Movable { // Java에선 new 타입이름() {익명 클래스 구현}
        override fun move() {        // Kotlin에선 object : 타입이름 {익명 클래스 구선}
            println("무브 무브")
        }

        override fun fly() {
            println("날아 날아")
        }
    })
}

private fun moveSomething(movable: Movable) {
    movable.move()
    movable.fly()
}
